manifests/gcp_marketplace/chart/kubeflow-pipelines/templates/pipeline.yaml

```diff
@@ -624,11 +624,7 @@ spec:
             # Following environment variables are only needed when using Cloud SQL and GCS.
             {{ if .Values.managedstorage.enabled }}
             - name: OBJECTSTORECONFIG_BUCKETNAME
-              {{ if .Values.managedstorage.databaseNamePrefix }}
-              value: '{{ printf "%s-%s" .Values.managedstorage.cloudsqlInstanceConnectionName .Values.managedstorage.databaseNamePrefix | replace ":" "-" | trunc 50 }}'
-              {{ else }}
-              value: '{{ printf "%s-%s" .Values.managedstorage.cloudsqlInstanceConnectionName .Release.Name | replace ":" "-" | trunc 50 }}'
-              {{ end }}
```

Comment:
I would prefer split the change into two PR. You have two in this PR
(1) fix the issue (argo bucket not respect managed storage)
(2) customize gcsBucketname
For (2), I remember Yang mentioned the reason is that he want to sync the DB name and Object storage name.
Need his LGTM in case (2) introduces other issue we didn't aware.

If Yang is OK with (2), then this PR requires touch schema.yaml to let UI feed in the gcsBucketName.

Reply:
Thanks! These are valid concerns.

I can also do just (1), but I didn't find a cleaner way than just duplicating the same {{if}} {{else}} in both places and make sure the complex logic for determining bucket name is the same.

It will be very helpful if you know other ways that can let me put the logic of determining bucket name in one place in helm chart, and then reference it in those sub charts that use it.


For (2), I wonder why we want to sync DB name and Object storage name?

Reply:
for (2), Yang mentioned with me some reason before. One reason I remember is that when user reinstall KFP but based on existing storage, it can save user's efforts to remember the pair (DB name and Storage buckage name) considering DB contains index to buckage. If user didn't well remember the pair or mis-match the pair, the error is pretty hard to be caught and understood by user.

for (1), I think a better solution is that we don't handle the logic inside helm but do it in application layer which is easier to share code logics while keep yaml be simple. It might require some corresponding code changes I didn't look into yet.

Reply:
thanks for the explanation.

> for (2), Yang mentioned with me some reason before. One reason I remember is that when user reinstall KFP but based on existing storage, it can save user's efforts to remember the pair (DB name and Storage buckage name) considering DB contains index to buckage. If user didn't well remember the pair or mis-match the pair, the error is pretty hard to be caught and understood by user.

The arguments sound reasonable. I think for clarity we should think about a step during marketplace install that we introduce to users which db/buckets are created for them.